probe-rs/src/probe/jlink/mod.rs

```diff
@@ -312,9 +315,25 @@ impl DebugProbe for JLink {
             // TODO: Add custom error
             return Err(DebugProbeError::ProbeCouldNotBeCreated);
         }
+        let jlink_handle = usb_devices.pop().unwrap().open()?;
+
+        let supported_protocols: Vec<WireProtocol> = if jlink_handle
```

Comment:
Can we maybe add a comment here, why this if is required? Because if anyone ever has to patch something, they'll ask themselves why we don't just always read the available interfaces :)
I mean it's really odd that it returns the JTag interface in case it can switch the interfaces but does not in the case it can't ;)
At least I would be curious why this is if I read that code.